WordPress/Sniffs/WP/DeprecatedParametersSniff.php

```diff
@@ -0,0 +1,317 @@
+			1 => array(
+				'value'   => 'true',
+				'version' => '2.8.1',
+			),
+		),
+		'get_delete_post_link' => array(
+			2 => array(
+				'value'   => '',
+				'version' => '3.0',
+			)
+		),
+		'is_email' => array(
+			2 => array(
+				'value'   => 'false',
+				'version' => '3.0',
+			),
+		),
+		'update_user_status' => array(
+			4 => array(
+				'value'   => 'null',
+				'version' => '3.0.2',
+			),
+		),
+		'update_blog_option' => array(
+			4 => array(
+				'value'   => 'null',
+				'version' => '3.1',
+			),
+		),
+		'update_blog_status' => array(
+			4 => array(
+				'value'   => 'null',
+				'version' => '3.1',
+			),
+		),
+		'wp_notify_postauthor' => array(
+			2 => array(
+				'value'   => 'null',
+				'version' => '3.8',
+			)
+		),
+		'wp_new_user_notification' => array(
+			2 => array(
+				'value'   => 'null',
+				'version' => '4.3.1',
+			)
+		),
+		'get_wp_title_rss' => array(
+			1 => array(
+				'value'   => '&#8211;',
+				'version' => '4.4',
+			)
+		),
+		'wp_title_rss' => array(
+			1 => array(
+				'value'   => '&#8211;',
+				'version' => '4.4',
+			),
+		),
+		'unregister_setting' => array(
+			4 => array(
+				'value'   => '',
+				'version' => '4.7',
+			),
+		),
+
+	);
+
+	/**
+	 * Process the parameters of a matched function.
+	 *
+	 * @since 0.11.0
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 *
+	 * @return void
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+		foreach ( $this->target_functions[ $matched_content ] as $position => $paramerter_args ) {
+
+			if ( ! isset( $parameters[ $position ] ) ) {
+				continue;
+			}
+
+			$is_dynamic_parameter = $this->phpcsFile->findNext( array( T_VARIABLE, T_STRING ), $parameters[ $position ]['start'], $parameters[ $position ]['end'], false, null, true );
+
+			$matched_parameter = $this->strip_quotes( $parameters[ $position ]['raw'] );
+
+			if ( ! $is_dynamic_parameter && $paramerter_args[ 'value' ] === $matched_parameter ) {
+				continue;
+			}
+
+			// 11th, 12th and 13th are not supported as they are an edge case.
+			switch ( $position ) {
+				case 1:
+					$pos_text = $position . 'st';
+					break;
+				case 2:
+					$pos_text = $position . 'nd';
+					break;
+				case 3:
+					$pos_text = $position . 'rd';
+					break;
+				default:
+					$pos_text = $position . 'th';
+					break;
+			}
+
+			$message = 'The %s parameter [%s] of %s() has been deprecated since WordPress version %s.';
+			$data    = array(
+				$pos_text,
+				$parameters[ $position ]['raw'],
+				$matched_content,
+				$paramerter_args['version'],
+			);
+
+			if ( isset( $paramerter_args['value'] ) ) {
+				$message .= ' Use "%s" instead.';
```

Comment:
Suggestion:
* If the parameter is the last parameter passed for the function, suggest not passing the parameter instead.
* In all other cases, suggest passing `null` instead as that will automatically fall back to the default used by the function and will potentially reduce sniff maintenance (and code maintenance for devs using the sniff) as it removes the need to sync the defaults with whatever is happening in core.